tests/functional/src/main/java/org/mule/functional/junit4/ArtifactFunctionalTestCase.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+
+package org.mule.functional.junit4;
+
+import static org.mule.functional.util.AnnotationUtils.getAnnotationAttributeFrom;
+import static org.mule.functional.util.AnnotationUtils.getAnnotationAttributeFromHierarchy;
+import org.mule.functional.junit4.runners.ArtifactClassLoaderRunnerConfig;
+import org.mule.functional.junit4.runners.ArtifactClassloaderRunnerContextAware;
+import org.mule.functional.junit4.runners.ArtifactClassloaderTestRunner;
+import org.mule.functional.junit4.runners.ClassLoaderIsolatedExtensionsManagerConfigurationBuilder;
+import org.mule.functional.junit4.runners.RunnerDelegateTo;
+import org.mule.runtime.core.api.config.ConfigurationBuilder;
+import org.mule.runtime.module.artifact.classloader.ArtifactClassLoader;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.junit.runner.RunWith;
+
+/**
+ * Base class for running {@link FunctionalTestCase} with class loader isolation that uses a {@link org.junit.runner.Runner}
+ * that classifies the classpath provided by IDE/surfire-maven-plugin and classifies the URLs using the maven dependency graph.
```

Comment:
I would add information about how to generate the dependencies file, explain how the dependencies are classified or add links to the classes that hold that info